docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,175 @@
+ - LdaNative
+   - No managed fallback, but builds successfully on non x86/x64 without changes.
+   - Large amount of work to have a managed fallback. This solution is about 3000 lines of code not including any code from the dependencies.
+   - This was created before we had hardware intrinsics so we used native code for performance.
+ - MatrixFactorizationNative
+   - No software fallback.
+   - Currently we are hardcoding the "USESSE" flag which requires x86/x64 SIMD commands. Removing this flag allows MatrixFactorizationNative to build for other platforms.
+   - Small amount of work required to change the build process and verify it's working.
+   - Large/Xlarge amount of work to have a managed fallback. This uses libmf, so we would have to not only port our code, but we would have to understand which parts of libmf are being used and port those as well.
+   - This was created before we had hardware intrinsics so we used native code for performance as well as take advantage of libmf.
+ - MklProxyNative
+   - Wrapper for Intel MKL. When Intel MKL is not present, this is not needed. However, the build is hardcoded to always compile this code.
+   - Small amount of work required to change the build process to exclude this as needed.
+ - SymSgdNative
+   - No managed fallback.
+   - Medium amount of work required to have a managed fallback. Only about 500 lines of code plus having to implement 4 vector operations.
+   - Small amount of work required port to other platforms. Only uses 4 Intel MKL methods which we can replace. I was only able to find 2 direct replacements on other architectures, but we could easily write our own for either those 2 or even all 4.
+   - This was created before we had hardware intrinsics so we used native code for performance as well as take advantage of IntelMKL.
+
+Of these 6, only LdaNative builds successfully for other architectures without changing anything.
+
+### 2.5 3rd Party Dependencies
+As mentioned above, there are several 3rd party packages that don't have support for non x86/x64 machines.
+ - LightGBM. LightGBM doesn't offer packages for non x86/x64. I was able to build the code for Arm64, but we would either have to build it ourselves, convince them to create more packages for us, or annotate that this doesn't work on non x86/64 machines.
+ - TensorFlow. The full version of TensorFlow only runs on x86/x64. There is a [lite](https://www.tensorflow.org/lite/guide/build_arm64) version that supports Arm64, and you can install it directly with python, but this isn't the full version so not all models will run. We would also have to verify if the C# library we use to interface with TensorFlow will work with the lite version.
+ - ONNX Runtime. ONNX Runtime doesn't have prebuilt packages for more than just x86/x64. It does support Arm, but we have to build it ourselves or get the ONNX Runtime team to package Arm assemblies. This is the same situation as with LightGBM.
```

Comment:
If we will need ONNX runtime and LightGBM for Arm* I suggest to reach out to them informally to see whether this is something they'd contemplate, whether it's part of their roadmap already, etc.

Reply:
We spoke with ONNX Runtime about it and they basically said that since there was no customer asks for it on arm that it wasn't on their soon to do list. Hopefully when ML.NET officially supports arm we can get the data needed to make that ask.

I haven't spoken with LightGBM about it. Do you have a contact for them? If not I can find something.